### PR TITLE
Fix Camera2D render target size bug & improve durability

### PR DESCRIPTION
**Pull request by pushfoo**
### Changes

1. Fix apparent bug where render target sizes are ignored by `Camera2D`
2. Use more durable keyword argument passing for dataclass arguments
3. Make `Camera2D.from_raw_data` inheritance-safe
4. Correct `Camera2D.viewport`'s docstring
5. Add tests for the changes above
6. Move the orthographic projector tests to an appropriately named file

#### Vec types
Once #2043 is merged, we should probably use its `Vec*` types liberally

#### Fancy / Named Tuple types?

Would `NamedTuple` or similar be better for internal `viewport` representation? Benefits I can see:
1. Provides named access which is cleaner and more legible
2. Named access may be faster for named props according to Ben's recent benchmarks for pyglet vectors (needs verification)

LRBT and similar types might be useful.

### Validation?

Currently we don't do any pyglet-style implicit unpacks or length checks. Maybe we should?

### CODE REVIEW

**Comment by pushfoo** [comment]
The remaining type issues in the GitHub based tests all appear to be GUI stuff.

**Comment by pushfoo** [comment]
TL;DR: Changes didn't break anything, it's just the GUI pyright issues again

The changes are responses to Discord feedback from DragonMoffon. GitHub based tests still report the same GUI breakage as before, and those are unrelated to this PR.
